pyogrio/_io.pyx

```diff
@@ -196,28 +205,33 @@ cdef void* ogr_open(const char* path_c, int mode, char** options) except NULL:
     else:
         flags |= GDAL_OF_READONLY
 
-
     try:
         # WARNING: GDAL logs warnings about invalid open options to stderr
         # instead of raising an error
-        ogr_dataset = exc_wrap_pointer(
-            GDALOpenEx(path_c, flags, NULL, <const char *const *>options, NULL)
-        )
-
-        return ogr_dataset
+        with stack_errors() as error_stack_checker:
```

Comment:
```suggestion
        with get_error_handler() as error_handler:
```

(to demonstrate increased clarity of above suggestions)

Reply:
I renamed the context handler to `capture_errors` instead of `get_error_handler` as I think this is a clearer description of the behaviour.

OK?